src/ParkChecklist/ParkChecklist.js

```diff
@@ -1,148 +1,221 @@
 import React, { useState, useEffect } from "react";
 import { useRoutes, useNavigate } from "react-router-dom";
 import axios from "axios";
-import PropTypes from 'prop-types';
+import PropTypes from "prop-types";
 import "./ParkChecklist.css";
 import page from "../resources/page.png";
 import SelectedStateParks from "../SelectedStateParks/SelectedStateParks";
+import ParkDetails from "../ParkDetails/ParkDetails";
 
 const ParkChecklist = ({ apiLink }) => {
-    const [parks, setParks] = useState([]);
-    const [checkedItems, setCheckedItems] = useState([]);
-    const [selectedState, setSelectedState] = useState("");
-    const [showResults, setShowResults] = useState(false);
-    const [currentPage, setCurrentPage] = useState(1);
-    const navigate = useNavigate();
-    const parksPerPage = 50;
-    const states = [
-        "AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "FL", "GA",
-        "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD",
-        "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ",
-        "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC",
-        "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"
-    ];
-
-    useEffect(() => {
-        const fetchData = async () => {
-            try {
-                const response = await axios.get(`${apiLink}&limit=500`);
-                setParks(response.data.data);
-            } catch (error) {
-                console.error("Error fetching data:", error);
-            }
-        };
-
-        fetchData();
-    }, [apiLink, selectedState, showResults, currentPage]);
-
-    const handleCheckboxChange = (parkId) => {
-        setCheckedItems((prevCheckedItems) => {
-            if (prevCheckedItems.includes(parkId)) {
-                return prevCheckedItems.filter((id) => id !== parkId);
-            } else {
-                return [...prevCheckedItems, parkId];
-            }
-        });
-    };
-
-    const handleStateChange = (event) => {
-        setSelectedState(event.target.value);
+  const [loading, setLoading] = useState(true);
+  const [parks, setParks] = useState([]);
+  const [parkID, setParkID] = useState(null);
+  const [checkedItems, setCheckedItems] = useState([]);
+  const [selectedState, setSelectedState] = useState("");
+  const [showResults, setShowResults] = useState(false);
+  const [foundPark, setFoundPark] = useState(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const navigate = useNavigate();
+  const parksPerPage = 50;
+  const states = [
+    "AL",
```

Comment:
was there a good reason for spreading it out? im fine with it either way, it just reduces our line count to compress it